internal/repository/target/constraints.go

```diff
@@ -0,0 +1,39 @@
+package target
+
+import (
+	"github.com/zitadel/zitadel/internal/eventstore"
+)
+
+const (
+	UniqueTarget    = "target"
+	DuplicateTarget = "Errors.Target.AlreadyExists"
+)
+
+func NewAddUniqueConstraints(name string) []*eventstore.UniqueConstraint {
+	return []*eventstore.UniqueConstraint{
+		eventstore.NewAddEventUniqueConstraint(
+			UniqueTarget,
+			name,
+			DuplicateTarget,
+		),
+	}
+}
+
+func NewRemoveUniqueConstraints(name string) []*eventstore.UniqueConstraint {
+	return []*eventstore.UniqueConstraint{
+		eventstore.NewRemoveUniqueConstraint(
+			UniqueTarget,
+			name,
+		),
+	}
+}
+
+func NewUpdateUniqueConstraints(oldName, name string) []*eventstore.UniqueConstraint {
+	return append(
+		append(
+			[]*eventstore.UniqueConstraint{},
+			NewRemoveUniqueConstraints(oldName)...,
+		),
+		NewAddUniqueConstraints(name)...,
+	)
+}
```

Comment:
i'd change the add and remove function to only return a single unique constraint and create a slice in the functions which use them, e.g. update or even remove the update at all, since it's only used once